src/perftest_resources.c

```diff
@@ -2228,47 +1642,98 @@ struct ibv_qp* ctx_qp_create(struct pingpong_context *ctx,
 		opcode = opcode_atomic_array[user_param->atomicType];
 		if(0);
 		else if (opcode == IBV_WR_ATOMIC_FETCH_AND_ADD)
-			attr.send_ops_flags |= IBV_QP_EX_WITH_ATOMIC_FETCH_AND_ADD;
+			attr_ex.send_ops_flags |= IBV_QP_EX_WITH_ATOMIC_FETCH_AND_ADD;
 		else if (opcode == IBV_WR_ATOMIC_CMP_AND_SWP)
-			attr.send_ops_flags |= IBV_QP_EX_WITH_ATOMIC_CMP_AND_SWP;
+			attr_ex.send_ops_flags |= IBV_QP_EX_WITH_ATOMIC_CMP_AND_SWP;
 	}
 	else {
 		opcode = opcode_verbs_array[user_param->verb];
 		if(0);
 		else if (opcode == IBV_WR_SEND)
-			attr.send_ops_flags |= IBV_QP_EX_WITH_SEND;
+			attr_ex.send_ops_flags |= IBV_QP_EX_WITH_SEND;
 		else if (opcode == IBV_WR_RDMA_WRITE)
-			attr.send_ops_flags |= IBV_QP_EX_WITH_RDMA_WRITE;
+			attr_ex.send_ops_flags |= IBV_QP_EX_WITH_RDMA_WRITE;
 		else if (opcode == IBV_WR_RDMA_READ)
-			attr.send_ops_flags |= IBV_QP_EX_WITH_RDMA_READ;
-	}
-	attr.pd = ctx->pd;
-	attr.comp_mask |= IBV_QP_INIT_ATTR_SEND_OPS_FLAGS | IBV_QP_INIT_ATTR_PD;
+			attr_ex.send_ops_flags |= IBV_QP_EX_WITH_RDMA_READ;
+	}
+	attr_ex.pd = ctx->pd;
+	attr_ex.comp_mask |= IBV_QP_INIT_ATTR_SEND_OPS_FLAGS | IBV_QP_INIT_ATTR_PD;
+	attr_ex.send_cq = attr.send_cq;
+	attr_ex.recv_cq = attr.recv_cq;
+	attr_ex.cap.max_send_wr = attr.cap.max_send_wr;
+	attr_ex.cap.max_send_sge = attr.cap.max_send_sge;
+	attr_ex.qp_type = attr.qp_type;
+	attr_ex.srq = attr.srq;
+	attr_ex.cap.max_inline_data = attr.cap.max_inline_data;
+	attr_ex.cap.max_recv_wr  = attr.cap.max_recv_wr;
+	attr_ex.cap.max_recv_sge = attr.cap.max_recv_sge;
 	#endif
 
 	if (user_param->work_rdma_cm) {
-		#if !defined(HAVE_IBV_WR_API)
-		if (rdma_create_qp(ctx->cm_id, ctx->pd, &attr)) {
-			fprintf(stderr, "Couldn't create rdma QP - %s\n", strerror(errno));
-		} else {
-			qp = ctx->cm_id->qp;
+		#ifdef HAVE_IBV_WR_API
+		if (!user_param->use_old_post_send)
+		{
+			if (rdma_create_qp_ex(ctx->cm_id, &attr_ex))
+			{
+				fprintf(stderr, "Couldn't create rdma new QP - %s\n", strerror(errno));
+			}
+			else
+			{
+				qp = ctx->cm_id->qp;
+			}
 		}
+		else
 		#endif
+			if (rdma_create_qp(ctx->cm_id, ctx->pd, &attr))
+			{
+				fprintf(stderr, "Couldn't create rdma old QP - %s\n", strerror(errno));
+			}
+			else
+			{
+				qp = ctx->cm_id->qp;
+			}
+
 	} else if (user_param->connection_type == SRD) {
 		#ifdef HAVE_SRD
 		qp = efadv_create_driver_qp(ctx->pd, &attr, EFADV_QP_DRIVER_TYPE_SRD);
```

Comment:
This is not rebased, but an if (use_old_post_send) should be added here as well to decide between efadv_create_driver_qp and efadv_create_qp_ex

Reply:
this will be covered by the backport of https://github.com/linux-rdma/perftest/pull/74